substra/sdk/backends/local/backend.py

```diff
@@ -739,8 +736,309 @@ def describe(self, asset_type, key):
+        )
+        compute_plan = self._db.add(compute_plan, exist_ok)
+
+        # go through the tuples sorted by rank
+        compute_plan = self.__execute_compute_plan(
+            spec,
+            compute_plan,
+            visited,
+            traintuples,
+            aggregatetuples,
+            compositetuples,
+            exist_ok,
+            spec_options
+        )
+        return compute_plan
+
+    def __get_all_tuples_compute_plan(self, spec: schemas.UpdateComputePlanSpec):
+        all_tuples = dict()
+        traintuples = dict()
+        if spec.traintuples:
+            for traintuple in spec.traintuples:
+                if traintuple.in_models_ids is not None:
+                    all_tuples[traintuple.traintuple_id] = traintuple.in_models_ids
+                else:
+                    all_tuples[traintuple.traintuple_id] = list()
+                traintuples[traintuple.traintuple_id] = traintuple
+
+        aggregatetuples = dict()
+        if spec.aggregatetuples:
+            for aggregatetuple in spec.aggregatetuples:
+                if aggregatetuple.in_models_ids is not None:
+                    all_tuples[aggregatetuple.aggregatetuple_id] = aggregatetuple.in_models_ids
+                else:
+                    all_tuples[aggregatetuple.aggregatetuple_id] = list()
+                aggregatetuples[aggregatetuple.aggregatetuple_id] = aggregatetuple
+
+        compositetuples = dict()
+        if spec.composite_traintuples:
+            for compositetuple in spec.composite_traintuples:
+                assert not compositetuple.out_trunk_model_permissions.public
+                all_tuples[compositetuple.composite_traintuple_id] = list()
+                if compositetuple.in_head_model_id is not None:
+                    all_tuples[compositetuple.composite_traintuple_id].append(
+                        compositetuple.in_head_model_id
+                    )
+                if compositetuple.in_trunk_model_id is not None:
+                    all_tuples[compositetuple.composite_traintuple_id].append(
+                        compositetuple.in_trunk_model_id
+                    )
+                compositetuples[compositetuple.composite_traintuple_id] = compositetuple
+
+        return all_tuples, traintuples, aggregatetuples, compositetuples
+
+    def __get_id_rank_in_compute_plan(self, type_, key, id_to_key):
+        tuple_ = self._db.get(schemas.Type.Traintuple, key)
+        id_ = next((k for k in id_to_key if id_to_key[k] == key), None)
+        return id_, tuple_.rank
+
+    def __execute_compute_plan(
+            self,
+            spec,
+            compute_plan,
+            visited,
+            traintuples,
+            aggregatetuples,
+            compositetuples,
+            exist_ok,
+            spec_options
+            ):
+        for id_, rank in {
+            id_: rank
+            for id_, rank
+            in sorted(visited.items(), key=lambda item: item[1])
+        }.items():
```

Comment:
Isn't this equivalent to `for id_, rank in sorted(visited.items(), key=lambda item: item[1]):` ?

Reply:
😅